Eatery/Controllers/BRB/BRBLoginViewController.swift

```diff
@@ -0,0 +1,338 @@
+//
+//  BRBLoginViewController.swift
+//  Eatery
+//
+//  Created by William Ma on 9/1/19.
+//  Copyright © 2019 CUAppDev. All rights reserved.
+//
+
+import NVActivityIndicatorView
+import UIKit
+
+// MARK: -
+
+private enum BRBAccountSettings {
+    
+    typealias LoginInfo = (netid: String, password: String)
+    
+    private static let saveLoginInfoKey = "save_login_info"
+    
+    static var saveLoginInfo: Bool {
+        get {
+            return UserDefaults.standard.bool(forKey: BRBAccountSettings.saveLoginInfoKey)
+        }
+        set {
+            UserDefaults.standard.set(newValue, forKey: BRBAccountSettings.saveLoginInfoKey)
+        }
+    }
+    
+    static func saveToKeychain(loginInfo: LoginInfo) {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        keychain["netid"] = loginInfo.netid as AnyObject
+        keychain["password"] = loginInfo.password as AnyObject
+    }
+    
+    static func removeKeychainLoginInfo() {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        keychain["netid"] = nil
+        keychain["password"] = nil
+        
+        BRBAccountSettings.saveLoginInfo = false
+    }
+    
+    static func loadFromKeychain() -> LoginInfo? {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        if let netid = keychain["netid"] as? String, let password = keychain["password"] as? String {
+            return (netid: netid, password: password)
+        } else{
+            return nil
+        }
+    }
+    
+}
+
+// MARK: - BRB Login View Controller Delegate
+
+protocol BRBLoginViewControllerDelegate: AnyObject {
+    
+    func loginViewController(_ loginViewController: BRBLoginViewController,
+                             didRequestLoginWithNetid netid: String,
+                             password: String)
+    
+}
+
+// MARK: - BRB Login View Controller
+
+class BRBLoginViewController: UIViewController {
+    
+    weak var delegate: BRBLoginViewControllerDelegate?
```

Comment:
can we make this private?

Reply:
Do you mean make the delegate private? We need write access to it externally (from `BRBViewController`)

Reply:
oh i see. could we pass pass in the `delegate` by creating a custom init for `BRBLoginViewController` and then change the `delegate` to be private? This way, the delegate will be set only once and we prevent the possibility of someone changing it later. We should make all delegates private because other classes shouldn't be accessing them.

Reply:
Seems like quite a bit of overhead for a pattern that's well established: The only owner of the `BRBLoginViewController`(the `BRBViewController`), has its reference to it private, so there's no chance of it being changed unintentionally outside of the owners file. 

Delegation is a well established pattern within Apple's frameworks, and it (almost) always works by assigning to a weak variable. Having that consistency with Apple's frameworks is also important. 

It's true that other classes shouldn't be accessing delegates, but I think that it's generally understood that delegates should only be assigned to. It seems that someone would have to be intentionally writing malicious code or just be uninformed to access the delegate in that way.

Reply:
Oh I think there was a bit of confusion. I totally agree that the delegate should be `weak`. I was just saying I think setting it to be `private` in addition would help. I also agree that in 99% of the cases, people probably won't write malicious code but I think additional layers of security are always good. The argument is similar for why we set view variables as private. It is understood that we shouldn't be setting these view variables outside of the class but making them `private` just helps to guarantee it. Passing the delegate through the init also guarantees that we only set the delegate once.